lib/python/eupath/GeneListExporter.py

```diff
@@ -0,0 +1,67 @@
+from handler_base.dataset_handler import DatasetHandler, ValidationException
+
+class GeneListExport(DatasetHandler):
+    """
+    This class is a specialized version of the Galaxy to EuPathDB dataset export tool.  This tool's
+    specialty is furnishing user's gene list data to EuPathDB.  As with all specialty export tools, this
+    tool implements 3 abstract classes.
+    """
+
+    # Name given to this type of dataset and to the expected file
+    GENE_LIST_TYPE = "GeneList"
+    GENE_LIST_VERSION = "1.0"
+    GENE_LIST_FILE = "genelist.txt"
+
+    # The validation script to be applied to the dataset files.  A failed validation should
+    # return in a system exit status of other than 0.
+    GENE_LIST_VALIDATION_SCRIPT = "validateGeneList"
+
+    def __init__(self, args):
+        """
+        Initializes the gene list export class with the parameters needed to accomplish the particular
+        type of export.
+        :param args: parameters provided from tool form
+        """
+        DatasetHandler.__init__(self,
+                                GeneListExport.GENE_LIST_TYPE,
+                                GeneListExport.GENE_LIST_VERSION,
+                                GeneListExport.GENE_LIST_VALIDATION_SCRIPT,
+                                args)
+
+        # For the gene list export, three parameters beyond generic 7 are required.
+        if len(args) < 8:
+            raise ValidationException("The tool was passed an insufficient numbers of arguments.")
+
+        # Data for the input given by the user
+        self._dataset_file_path = args[7]
```

Comment:
In Galaxy, I think there may be some guarantees about the format of the file being exported.  If the user is sending us a gene list from a step, then that is also handled (albeit client-side), but the user could literally upload any file they want here- wondering if/where we are validating at least that this is a text file with a column of words?  If we're not, we probably should.

Reply:
Got it, I made some README updates, I'll update this PR with some validation as well and possibly README updates as well